platform/qt/src/qmapboxgl.cpp

```diff
@@ -991,6 +1194,23 @@ void QMapboxGL::addCustomLayer(const QString &id,
             before ? mbgl::optional<std::string>(before) : mbgl::optional<std::string>());
 }
 
+/*!
+    Adds a style layer to the map as specified by \l
+    {https://www.mapbox.com/mapbox-gl-style-spec/#root-layers}{the vector tiles specification} with
+    \a params.
+
+    This example shows how to add a layer that will be used to show a route line on the map. Note
+    that nothing will be drawn until we set paint properties using setPaintProperty().
```

Comment:
Note that the source must be defined before the layer is added. As I understand it, changing a source after adding a layer that depends on the source results in undefined behavior.